File: my-folder/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 1:
            return False
        hm = { "}" : "{", "]" : "[", ")" : "("}
        stack = [s[0]]
        for char in s[1: ]:
            if char in '}])' and  stack and stack[-1] == hm[char]:
                stack.pop()
            elif char in '}])' and stack and stack[-1] != hm[char]:
                return False
            else:
                stack.append(char)

        return True if not stack else False

        

File: my-folder/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character, Character> hmap = new HashMap<>();
        hmap.put(')', '(');
        hmap.put('}', '{');
        hmap.put(']', '[');


        for(char curr : s.toCharArray()){
            if(curr == '(' || curr == '{' || curr == '['){
                stack.add(curr);
            }
            else{
                if(stack.isEmpty() || stack.peek() != hmap.get(curr)){
                    return false;
                }
                else{
                    stack.pop();
                } 
            }
        }

        return stack.isEmpty() ? true : false;
    }
}
